googleapiclient/discovery_cache/__init__.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2014 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Caching utility for the discovery document."""
+
+
+from __future__ import absolute_import
+
+import logging
+import datetime
+
+
+DISCOVERY_DOC_MAX_AGE = 60 * 60 * 24  # 1 day
+
+
+def autodetect():
+  """Detects an appropriate cache module and returns it.
+
+  Returns:
+    An object with the following methods; get(url), set(url, content).
```

Comment:
Describe the type of the returned value. Also, since this function can return either None or a value other than None, put explicit return statements at the end of every code path through its body. Never mix "explicitly return a non-None value" and "return None by default after falling off the bottom of the method body" semantics.